clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -576,10 +576,9 @@ public boolean maybeHandleDisconnection(ClientRequest request) {
         }
 
         @Override
-        public boolean maybeHandleCompletedReceive(ClientRequest req, long now, Struct body) {
-            short apiKey = req.request().header().apiKey();
-            if (apiKey == ApiKeys.METADATA.id && req.isInitiatedByNetworkClient()) {
-                handleResponse(req.request().header(), body, now);
+        public boolean maybeHandleCompletedReceive(ClientRequest req, long now, AbstractResponse response) {
```

Comment:
Perhaps we could rename this method to sth like maybeHandleCompletedMetadataResponse to make it clear?